src/Helper/RulesNormalizer.php

```diff
@@ -82,6 +82,10 @@ private static function prepareRulesArray(
             return [$rules];
         }
 
+        if (is_callable($rules)) {
```

Comment:
Maybe modify the condition above instead?

```php
if ($rules instanceof RuleInterface || $is_callable($rules)) {
    return [$rules];
}
```

And the `Callback` conversion is already performed here for single rule:

https://github.com/yiisoft/validator/blob/9ae20f808ddd706ac3cfb9ac74c8c11af1123292/src/Helper/RulesNormalizer.php#L115-L117

Reply:
done